### PR TITLE
Introduce Spatial Plugin

### PR DESCRIPTION
**Pull request by talevy**
This commit introduces a skeleton xpack Spatial plugin that will be filled-in
with some new licensed features coming to Geo-Spatial!

This will just be merged into `master`, until the first feature merged in the spatial plugin 
will be merged. Then we can back-port this PR to 7.x prior to merging in any future 7x features

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java [hunk 1]
```diff
@@ -121,6 +121,12 @@ private XPackSettings() {
     /** Setting for enabling or disabling vectors. Defaults to true. */
     public static final Setting<Boolean> VECTORS_ENABLED = Setting.boolSetting("xpack.vectors.enabled", true, Setting.Property.NodeScope);
 
+    /**
+     * Setting for enabling or disabling geo features. Defaults to true.
+     */
+    public static final Setting<Boolean> GEO_ENABLED = Setting.boolSetting("xpack.geo.enabled", true,
```
Comment (talevy) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java]: should we have this setting? Do we foresee anyone wanting to disable any of this?

**Comment by iverase** [comment]
just a generic question: Is this really Geo or should be called Spatial? My point here is that if in this package we will introduce XYShape as well, then Geo is too specific.

**Comment by talevy** [comment]
@iverase that is a good point, I think that may be the better name. I'll check with some others as well

**Comment by talevy** [comment]
run elasticsearch-ci/1

File: x-pack/plugin/spatial/build.gradle [hunk 2]
```diff
@@ -0,0 +1,21 @@
+evaluationDependsOn(xpackModule('core'))
```
[the remark this answers is not in the document]
Comment (talevy) [reply x-pack/plugin/spatial/build.gradle]: not sure, I do not see many others doing this. happy to move towards doing this if we feel this is appropriate

File: x-pack/plugin/spatial/build.gradle [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+dependencies {
+    compileOnly project(path: xpackModule('core'), configuration: 'default')
+    testCompile project(path: xpackModule('core'), configuration: 'testArtifacts')
+    if (isEclipse) {
+        testCompile project(path: xpackModule('core-tests'), configuration: 'testArtifacts')
+    }
+}
+
+integTest.enabled = false
```
Comment (rjernst) [inline x-pack/plugin/spatial/build.gradle]: please include the comment that accompanied this in other build.gradle files

File: x-pack/plugin/spatial/build.gradle [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+apply plugin: 'elasticsearch.esplugin'
+
+esplugin {
+    name 'spatial'
+    description 'A plugin for Basic Spatial features'
+    classname 'org.elasticsearch.xpack.spatial.Spatial'
+    extendedPlugins = ['x-pack-core']
+}
+archivesBaseName = 'x-pack-spatial'
```
Comment (rjernst) [inline x-pack/plugin/spatial/build.gradle]: This shouldn't be necessary

File: x-pack/plugin/spatial/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (rjernst) [reply x-pack/plugin/spatial/build.gradle]: A copyright header is not necessary in our build.gradle files.

**Review by rjernst — COMMENTED** [review]
I don't think we need the "enabled"? We recently added voting-only without it, and I don't htink it is necessary for most things.

**Comment by talevy** [comment]
thanks Ryan, mind taking another look?

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**